### ISSUE TITLE
label-warn for other modules

### ISSUE THREAD

**Issue by noctuid** [polybar/polybar#570]
I think it would make sense to be able to have warning thresholds for the memory, cpu, and battery modules as well.

**Comment by jaagr** [comment]
You could already achieve the same behavior by using `<ramp>` if the module have support for it. For example:

~~~ dosini
[module/cpu]
type = internal/cpu
format = <ramp-load>

ramp-load-0 = low
ramp-load-0-foreground = #0f0
ramp-load-1 = medium
ramp-load-1-foreground = #ff0
ramp-load-2 = high
ramp-load-2-foreground = #f00

# or

ramp-load-0 = normal
ramp-load-1 = normal
ramp-load-2 = normal
ramp-load-3 = warning

**Comment by noctuid** [comment]
I tried that initially, but it doesn't seem possible to use `%percentage%` outside of the label. Also the memory module doesn't have `<ramp>`, and it would be more precise and less verbose to be able to use a specific value for a threshold.

**Comment by boydthomson** [comment]
noctuid, did you find a way to achieve this?  I'm looking for the same feature.  Would be great to have greater visual distraction when my RAM usage is getting up there.

**Comment by juliancoffee** [comment]
It is a great idea. I implemented it for memory in #2142, and want to implement it for CPU as well.

**Comment by derrgill** [comment]
Love this idea, I also would like to be able to set arbitrary warning thresholds (e.g. RAM and swap percentages, even network download/upload speeds, and perhaps eventually disk read/write speeds if a disk I/O module is implemented) and thus conditional label warn foreground colors. Pretty much everything in my polybar is just colored an unobtrusive white and only changes color IF something actually changes, e.g., I have a custom script module that only changes to red if my ISP has changed my WAN ipv4 and/or ipv6 address(es).

**Comment by DRKblade** [comment]
I'm working on this feature. It has been implemented succesfully on the memory, cpu and filesystem modules, only the battery module left and I will upload the code.

But I was wondering, do you need a warning on the alsa/pulseaudio module? Maybe for protection from loud sounds? If so, I will implement this feature on those modules too.

**Comment by patrick96** [comment]
Implemented in #2199
